effect/ref.py

```diff
@@ -0,0 +1,60 @@
+from characteristic import attributes
+
+from ._base import Effect
+from ._dispatcher import TypeDispatcher
+from ._sync import sync_performer
+
+
+class ERef(object):
+    """
+    An effectful mutable variable.
+
+    Compare to Haskell's ``IORef`` or Clojure's ``atom``.
```

Comment:
Very nice explanation!  Yes, this very clear!  I was just suggesting that this be documented somewhere, either in this docstring, or in the module docstring, or in a wiki or readme some place, because someone just coming across this class might wonder (as I did) why it existed.  The comment above I think can pretty much be copy-pasted - it's pretty awesome.

Thanks for taking the time to explain!